temporal-sdk/src/test/java/io/temporal/worker/CleanWorkerShutdownTest.java

```diff
@@ -214,141 +98,50 @@ public void testShutdownNow() throws ExecutionException, InterruptedException {
-    GetWorkflowExecutionHistoryResponse result =
-        service.blockingStub().getWorkflowExecutionHistory(request);
-    List<HistoryEvent> events = result.getHistory().getEventsList();
-    boolean found = false;
-    for (HistoryEvent e : events) {
-      if (e.getEventType() == EventType.EVENT_TYPE_ACTIVITY_TASK_COMPLETED) {
-        found = true;
-        Payloads ar = e.getActivityTaskCompletedEventAttributes().getResult();
-        String r =
-            DataConverter.getDefaultInstance()
-                .fromPayloads(0, Optional.of(ar), String.class, String.class);
-        assertEquals("workershutdown", r);
-      }
-    }
-    assertTrue("Contains ActivityTaskCompleted", found);
-    if (useExternalService) {
-      service.shutdownNow();
-      service.awaitTermination(10, TimeUnit.MINUTES);
-    }
-  }
+  public static class TestWorkflowImpl implements TestWorkflow1 {
 
-  public static class TestWorkflowImpl implements TestWorkflowReturnString {
-
-    private final NoArgsReturnsStringActivity activities =
+    private final TestActivity1 activities =
         Workflow.newActivityStub(
-            NoArgsReturnsStringActivity.class,
+            TestActivity1.class,
             ActivityOptions.newBuilder()
                 .setScheduleToCloseTimeout(Duration.ofSeconds(100))
                 .build());
 
     @Override
-    public String execute() {
-      return activities.execute();
+    public String execute(String now) {
+      return activities.execute(now);
     }
   }
 
-  public static class ActivitiesImpl implements NoArgsReturnsStringActivity {
-    private final CompletableFuture<Boolean> started;
-
-    public ActivitiesImpl(CompletableFuture<Boolean> started) {
-      this.started = started;
-    }
-
-    @Override
-    public String execute() {
-      try {
-        started.complete(true);
-        Thread.sleep(1000);
-      } catch (InterruptedException e) {
-        Thread.currentThread().interrupt();
-        return "interrupted";
-      }
-      return "completed";
-    }
-  }
+  public static class ActivitiesImpl implements TestActivity1 {
 
-  public static class HeartbeatingActivitiesImpl implements NoArgsReturnsStringActivity {
-    private final CompletableFuture<Boolean> started;
+    private final CountDownLatch shutdownLatch;
+    private final CountDownLatch shutdownNowLatch;
 
-    public HeartbeatingActivitiesImpl(CompletableFuture<Boolean> started) {
-      this.started = started;
+    public ActivitiesImpl(CountDownLatch shutdownLatch, CountDownLatch shutdownNowLatch) {
+      this.shutdownLatch = shutdownLatch;
+      this.shutdownNowLatch = shutdownNowLatch;
     }
 
     @Override
-    public String execute() {
+    public String execute(String now) {
       try {
-        started.complete(true);
-        Thread.sleep(1000);
-        Activity.getExecutionContext().heartbeat("foo");
-      } catch (ActivityWorkerShutdownException e) {
-        return "workershutdown";
+        if (now == null) {
```

Comment:
Please move these latches code block up from the `try{} catch` to improve readability.
I believe `CountDownLatch#countDown` shouldn't throw InterruptedException

Reply:
Moved, but it does not throw the InterruptedException